### ISSUE TITLE
Regular reporting callbacks during evolution

### ISSUE THREAD

**Issue by mmastrac** [basvanwesting/genetic-algorithm#4]
When running longer evolve processes, it would be useful to specify a custom callback that could log the state of the evolution. I _think_ it's possible to replicate this by setting `with_max_stale_generations` lower and re-running `.call`, but ideally `report_round` would be pluggable.

I think an extension would also work here but it feels like a pluggable reporter would be more fitting given the design of the library.

**Comment by basvanwesting** [comment]
I have a pluggable reporting implementation on main and will wrap this up for a new release.
Example below:

```rust
use genetic_algorithm::strategy::evolve::prelude::*;

#[derive(Clone)]
pub struct CustomReporter { pub period: usize }
impl EvolveReporter for CustomReporter {
    type Genotype = BinaryGenotype;

    fn on_new_generation(&mut self, state: &EvolveState<Self::Genotype>) {
        if state.current_generation() % self.period == 0 {
            println!(
                "periodic - current_generation: {}, best_generation: {}, current_population_size: {}",
                state.current_generation(),
                state.best_generation(),
                state.population.size(),
            );
        }
    }

    fn on_new_best_chromosome(&mut self, state: &EvolveState<Self::Genotype>) {
        println!(
            "new best - generation: {}, fitness_score: {:?}, genes: {:?}, population_size: {}",
            state.current_generation(),
            state.best_fitness_score(),
            state.best_chromosome_as_ref().map(|c| &c.genes),
            state.population.size(),
        );
    }
}
```

**Comment by basvanwesting** [comment]
All feedback is very welcome of course

**Comment by basvanwesting** [comment]
Released v0.8.0 with pluggable reporting implementation

**Comment by basvanwesting** [comment]
Released v0.8.0
